### PR TITLE
Add spell check pin changes

### PR DESCRIPTION
**Pull request by danieljurek**
Changes from PR: https://github.com/Azure/azure-sdk-for-js/pull/17142/checks

### CODE REVIEW

File: eng/common/spelling/Invoke-Cspell.ps1 [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+  [Parameter()]
+  [switch] $LeavePackageInstallCache,
+
+  [Parameter()]
+  [switch] $Test
+)
+
+Set-StrictMode -Version 3.0
+
+if ((Get-Command npm | Measure-Object).Count -eq 0) {
```
Comment (weshaggard) [inline eng/common/spelling/Invoke-Cspell.ps1:75]:
```suggestion
if (!(Get-Command npm -ErrorAction SilentlyContinue)) {
```

File: eng/common/spelling/Invoke-Cspell.ps1 [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+# Set the temporary config file with the mutated configuration. The temporary
+# location is used to configure the command and the original file remains
+# unchanged.
+Write-Host "Setting config in: $CSpellConfigPath"
+Set-Content `
+  -Path $CSpellConfigPath `
+  -Value (ConvertTo-Json $cspellConfig -Depth 100)
+
+# Before chaning the run location, resolve paths specified in parameters
```
Comment (weshaggard) [inline eng/common/spelling/Invoke-Cspell.ps1:162]:
```suggestion
# Before changing the run location, resolve paths specified in parameters
```

**Review by weshaggard — APPROVED** [review]

File: eng/common/pipelines/templates/steps/check-spelling.yml [hunk 3]
```diff
@@ -14,13 +16,25 @@ parameters:
 
 steps:
   - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
+    - task: NodeTool@0
+      inputs:
+        versionSpec: 16.x
+      displayName: Use Node.js 16.x
+
+    # NPM version installed on some machines is very outdated and does not
```
Comment (weshaggard) [inline eng/common/pipelines/templates/steps/check-spelling.yml:24]: As we talked lets try to avoid this and use npx if possible.

**Comment by danieljurek** [comment]
@weshaggard -- Subbed in `npx` for `npm exec`

**Pull request merged**